posts/atomic-design-bootstrap.md

```diff
@@ -0,0 +1,154 @@
+
+## Utilities
+
+Some elements are use as utilities, they usually have a lower graphical value than `Components` and are used by composition..
+
+As an example, we will use [text](https://getbootstrap.com/docs/4.4/utilities/text).
+
+This utility let you do generics actions like:
+
+* alignment
+* prevent line return and truncate
+* transform to lowercase, uppercase …
+* change weight and use italic
+* change font style to monospace
+
+Some rules are more generic than others: where alignment is common and will have a small effect on style, the specific font choice for something like monospace may have more impacts:
+
+```css
+.text-monospace {
+    font-family: SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace;
+}
+```
+
+May be you can think it is not an important effect, but, depending on the utilities, this type of effect may vary, this is the point of [colors](https://getbootstrap.com/docs/4.4/utilities/colors) which has its own semantic and refers to _Bootstrap_ branding.
+
+You must be pragmatic to choose utilities.
+
+# Go further with Sass
+
+Extends and customize _Bootstrap_ can be done using  [Sass (SCSS)](https://github.com/twbs/bootstrap) [sources](https://sass-lang.com). It is possible to change container size depending on window size, colors, and there is many other exposed variables using `!default` Sass principle. Moreover, file fragmentation let you use parts of _Bootstrap_ like use only the grid system for example by importing the Sass file you want to use.
+
+In this case, you can use them for:
+
+* `Components` : Change colors for perception
+* `grid system` : Change columns number to fit what you need
+
+More information on what is available to extends _Bootstrap_ are on the [Theming Bootstrap](https://getbootstrap.com/docs/4.4/getting-started/theming) page.
+
+I recommend you to not override too many properties of _Bootstrap_ because your coupling can impact your maintainability and version upgrades.
+
+# Conclusion
+
+There is a lot of way to use _Bootstrap_ in an _Atomic Design_ project. With the experience and depending on the context, we learn how to not override, not use build-in _Bootstrap_ components without think about how they will be used and how they are perceived. Sometimes it is not needed to add the framework because it may not add value.
+
+Personally, I prefer develop new components by myself to add the best graphical value I can to the application.
+
+In another way, be pragmatic let structure components into atoms, molecules and organisms based on granularity and the way to assemble them. My usage of _Bootstrap_ today, when I am in a project with this kind of framework, is only with some utilities like `d-none` or alignments like `text-right`. I also use the `grid-system` but as a source of inspiration because most of the time the graphical identity doesn't match what propose _Bootstrap_.
```

Comment:
be pragmatic,*

>  "be pragmatic, let structure components into atoms [...]" 

Not sure what you mean here, a word (verb?) might be missing here


doesn't match with what _Bootstrap_ offers*

Reply:
So i changed it to :

```
In another way, be pragmatic helps to structure components into atoms, molecules and organisms based on granularity and the way to assemble them.
```

The global idea is not to stay rigid with three levels of granularity.